### PR TITLE
UNDERTOW-1229 HttpClientConnection should use equalToString and prefix header constants

### PR DESCRIPTION
**Pull request by frapex**
Function equalToString(String) is more efficient than equals(HttpString) as it does not require an additional object allocation.

Prefix header constants to make clear we are dealing with a header, not with some other constant such as CLOSE_REQ, CLOSED.

### CODE REVIEW

**Comment by undertow-pull-request** [comment]
**Linux** [Build 2520](https://ci.wildfly.org/viewLog.html?buildId=79333&buildTypeId=UT_MASTER_LINUX) outcome was **FAILURE** using a merge of d982b4d23342bfb7f0e4b2ee6c48df2b62e03756
Summary: Tests failed: 3, passed: 3063, ignored: 575 Build time: 00:20:39
### Failed tests
```
io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge[proxy][ajp]: 	at io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge(MultiPartTestCase.java:187)
------- Stderr: -------
01:13:37,277 DEBUG (XNIO-1 task-14) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /servletContext/2, authentication required: false
01:13:37,277 DEBUG (XNIO-1 task-14) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@6adf39 for /servletContext/2
01:13:37,277 DEBUG (XNIO-1 task-14) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /servletContext/2
01:13:37,279 ERROR (XNIO-1 I/O-8) [io.undertow.proxy] <ProxyHandler.java:660> UT005028: Proxy request to /servletContext/2 failed: java.nio.channels.ClosedChannelException
	at io.undertow.client.ajp.AjpClientConnection$ClientReceiveListener.handleEvent(AjpClientConnection.java:359)
	at io.undertow.client.ajp.AjpClientConnection$ClientReceiveListener.handleEvent(AjpClientConnection.java:351)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:921)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:902)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
	at org.xnio.nio.NioSocketConduit.handleReady(NioSocketConduit.java:88)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)

01:13:37,280 DEBUG (XNIO-1 I/O-8) [io.undertow.client.ajp.AjpClientConnection] <AjpClientConnection.java:116> connection to localhost/127.0.0.1:8888 closed


io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge[proxy][http2]: 	at io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge(MultiPartTestCase.java:187)
------- Stderr: -------
01:16:38,205 DEBUG (XNIO-1 task-1) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /servletContext/2, authentication required: false
01:16:38,206 DEBUG (XNIO-1 task-1) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@784f7b for /servletContext/2
01:16:38,206 DEBUG (XNIO-1 task-1) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /servletContext/2
01:16:38,207 ERROR (XNIO-1 I/O-4) [io.undertow.proxy] <ProxyHandler.java:660> UT005028: Proxy request to /servletContext/2 failed: java.io.IOException: UT000103: Http2 stream was reset
	at io.undertow.client.http2.Http2ClientConnection$Http2ReceiveListener.handleEvent(Http2ClientConnection.java:456)
	at io.undertow.client.http2.Http2ClientConnection$Http2ReceiveListener.handleEvent(Http2ClientConnection.java:397)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:921)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:902)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
	at io.undertow.protocols.ssl.SslConduit$SslReadReadyHandler.readReady(SslConduit.java:1145)
	at org.xnio.nio.NioSocketConduit.handleReady(NioSocketConduit.java:88)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)


io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge[proxy][http2-clear-upgrade]: 	at io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge(MultiPartTestCase.java:187)
------- Stderr: -------
01:18:02,784 DEBUG (XNIO-1 task-24) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /servletContext/2, authentication required: false
01:18:02,785 DEBUG (XNIO-1 task-24) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@14d4d23 for /servletContext/2
01:18:02,785 DEBUG (XNIO-1 task-24) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /servletContext/2
01:18:02,785 ERROR (XNIO-1 I/O-4) [io.undertow.proxy] <ProxyHandler.java:660> UT005028: Proxy request to /servletContext/2 failed: java.io.IOException: UT000103: Http2 stream was reset
	at io.undertow.client.http2.Http2ClientConnection$Http2ReceiveListener.handleEvent(Http2ClientConnection.java:456)
	at io.undertow.client.http2.Http2ClientConnection$Http2ReceiveListener.handleEvent(Http2ClientConnection.java:397)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:921)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:902)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
	at org.xnio.nio.NioSocketConduit.handleReady(NioSocketConduit.java:88)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)


```

**Comment by undertow-pull-request** [comment]
**Windows** [Build 2274](https://ci.wildfly.org/viewLog.html?buildId=79334&buildTypeId=UNDERTOW_MasterWindows) outcome was **FAILURE** using a merge of d982b4d23342bfb7f0e4b2ee6c48df2b62e03756
Summary: Tests failed: 4 (1 new), passed: 2984, ignored: 653 Build time: 00:37:58
### Failed tests
```
io.undertow.server.handlers.ReceiverTestCase.testAsyncReceiveWholeBytesFailed[proxy][http2]: java.lang.AssertionError
	at io.undertow.server.handlers.ReceiverTestCase.testAsyncReceiveWholeBytesFailed(ReceiverTestCase.java:194)
------- Stderr: -------
09:05:14,059 DEBUG (XNIO-1 I/O-5) [io.undertow.server.handlers.proxy] <ProxyHandler.java:547> Sending request ClientRequest{path='/fullbytes', method=POST, protocol=HTTP/1.1} to target localhost/127.0.0.1:8888 for exchange HttpServerExchange{ POST /fullbytes request {Content-Length=[130000100], Host=[localhost]} response {}}
09:05:14,062 DEBUG (XNIO-1 I/O-5) [io.undertow.server.handlers.proxy] <ProxyHandler.java:564> Sent request ClientRequest{path='/fullbytes', method=POST, protocol=HTTP/1.1} to target 127.0.0.1 for exchange HttpServerExchange{ POST /fullbytes request {Content-Length=[130000100], Host=[localhost]} response {}}
09:05:14,317 ERROR (XNIO-1 I/O-5) [io.undertow.proxy] <ProxyHandler.java:660> UT005028: Proxy request to /fullbytes failed: java.nio.channels.ClosedChannelException
	at io.undertow.client.http2.Http2ClientConnection$1.handleEvent(Http2ClientConnection.java:108)
	at io.undertow.client.http2.Http2ClientConnection$1.handleEvent(Http2ClientConnection.java:100)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameCloseListener.handleEvent(AbstractFramedChannel.java:1034)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameCloseListener.handleEvent(AbstractFramedChannel.java:948)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.protocols.ssl.SslConduit$SslReadReadyHandler.terminated(SslConduit.java:1186)
	at org.xnio.nio.NioSocketConduit.readTerminated(NioSocketConduit.java:360)
	at org.xnio.nio.NioSocketStreamConnection.notifyReadClosed(NioSocketStreamConnection.java:148)
	at org.xnio.Connection.close(Connection.java:139)
	at org.xnio.IoUtils.safeClose(IoUtils.java:134)
	at io.undertow.protocols.ssl.SslConduit.closed(SslConduit.java:1021)
	at io.undertow.protocols.ssl.SslConduit.close(SslConduit.java:1092)
	at io.undertow.protocols.ssl.UndertowSslConnection.closeAction(UndertowSslConnection.java:149)
	at org.xnio.Connection.close(Connection.java:132)
	at org.xnio.IoUtils.safeClose(IoUtils.java:134)
	at io.undertow.server.protocol.framed.AbstractFramedChannel.close(AbstractFramedChannel.java:789)
	at org.xnio.IoUtils.safeClose(IoUtils.java:134)
	at io.undertow.protocols.http2.Http2Channel.sendGoAway(Http2Channel.java:824)
	at io.undertow.protocols.http2.Http2Channel.sendGoAway(Http2Channel.java:801)
	at io.undertow.client.http2.Http2ClientConnection.close(Http2ClientConnection.java:313)
	at org.xnio.IoUtils.safeClose(IoUtils.java:134)
	at io.undertow.server.handlers.proxy.ProxyHandler$IoExceptionHandler.handleException(ProxyHandler.java:788)
	at org.xnio.ChannelListeners.invokeChannelExceptionHandler(ChannelListeners.java:126)
	at io.undertow.util.Transfer$TransferListener.handleEvent(Transfer.java:218)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedStreamSinkChannel$1.run(AbstractFramedStreamSinkChannel.java:214)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$3.run(AbstractFramedChannel.java:230)
	at org.xnio.nio.WorkerThread.safeRun(WorkerThread.java:582)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:466)

09:05:14,318 DEBUG (XNIO-1 I/O-7) [io.undertow.request.io] <AbstractFramedChannel.java:813> Marking reads broken on channel Http2Channel peer /127.0.0.1:51198 local 0.0.0.0/0.0.0.0:8888[ No Receiver [] -- [] -- []]: java.nio.channels.ClosedChannelException
	at io.undertow.server.protocol.framed.AbstractFramedChannel.markReadsBroken(AbstractFramedChannel.java:813)
	at io.undertow.server.protocol.framed.AbstractFramedChannel.receive(AbstractFramedChannel.java:470)
	at io.undertow.server.protocol.http2.Http2ReceiveListener.handleEvent(Http2ReceiveListener.java:109)
	at io.undertow.server.protocol.http2.Http2ReceiveListener.handleEvent(Http2ReceiveListener.java:60)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:921)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:902)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
	at io.undertow.protocols.ssl.SslConduit$SslReadReadyHandler.readReady(SslConduit.java:1145)
	at io.undertow.protocols.ssl.SslConduit$1.run(SslConduit.java:168)
	at org.xnio.nio.WorkerThread.safeRun(WorkerThread.java:582)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:466)


io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge[proxy][ajp]: 	at io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge(MultiPartTestCase.java:187)
------- Stderr: -------
09:16:20,093 DEBUG (XNIO-1 task-30) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /servletContext/2, authentication required: false
09:16:20,095 DEBUG (XNIO-1 task-30) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@7c16b6 for /servletContext/2
09:16:20,096 DEBUG (XNIO-1 task-30) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /servletContext/2
09:16:20,099 ERROR (XNIO-1 I/O-6) [io.undertow.proxy] <ProxyHandler.java:660> UT005028: Proxy request to /servletContext/2 failed: java.nio.channels.ClosedChannelException
	at io.undertow.client.ajp.AjpClientConnection$ClientReceiveListener.handleEvent(AjpClientConnection.java:359)
	at io.undertow.client.ajp.AjpClientConnection$ClientReceiveListener.handleEvent(AjpClientConnection.java:351)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:921)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:902)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
	at org.xnio.nio.NioSocketConduit.handleReady(NioSocketConduit.java:88)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)

09:16:20,106 DEBUG (XNIO-1 I/O-6) [io.undertow.client.ajp.AjpClientConnection] <AjpClientConnection.java:116> connection to localhost/127.0.0.1:8888 closed


io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge[proxy][http2]: 	at io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge(MultiPartTestCase.java:187)
------- Stderr: -------
09:22:54,577 DEBUG (XNIO-1 task-15) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /servletContext/2, authentication required: false
09:22:54,578 DEBUG (XNIO-1 task-15) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@1a775a2 for /servletContext/2
09:22:54,578 ERROR (XNIO-1 I/O-5) [io.undertow.proxy] <ProxyHandler.java:660> UT005028: Proxy request to /servletContext/2 failed: java.io.IOException: UT000103: Http2 stream was reset
	at io.undertow.client.http2.Http2ClientConnection$Http2ReceiveListener.handleEvent(Http2ClientConnection.java:456)
	at io.undertow.client.http2.Http2ClientConnection$Http2ReceiveListener.handleEvent(Http2ClientConnection.java:397)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:921)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:902)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
	at io.undertow.protocols.ssl.SslConduit$SslReadReadyHandler.readReady(SslConduit.java:1145)
	at org.xnio.nio.NioSocketConduit.handleReady(NioSocketConduit.java:88)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)

09:22:54,578 DEBUG (XNIO-1 task-15) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /servletContext/2


io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge[proxy][http2-clear-upgrade]: 	at io.undertow.servlet.test.multipart.MultiPartTestCase.testMultiPartRequestToLarge(MultiPartTestCase.java:187)
------- Stderr: -------
09:24:49,820 DEBUG (XNIO-1 task-13) [io.undertow.request.security] <SecurityContextImpl.java:86> Attempting to authenticate /servletContext/2, authentication required: false
09:24:49,820 ERROR (XNIO-1 I/O-8) [io.undertow.proxy] <ProxyHandler.java:660> UT005028: Proxy request to /servletContext/2 failed: java.io.IOException: UT000103: Http2 stream was reset
	at io.undertow.client.http2.Http2ClientConnection$Http2ReceiveListener.handleEvent(Http2ClientConnection.java:456)
	at io.undertow.client.http2.Http2ClientConnection$Http2ReceiveListener.handleEvent(Http2ClientConnection.java:397)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:921)
	at io.undertow.server.protocol.framed.AbstractFramedChannel$FrameReadListener.handleEvent(AbstractFramedChannel.java:902)
	at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
	at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
	at org.xnio.nio.NioSocketConduit.handleReady(NioSocketConduit.java:88)
	at org.xnio.nio.WorkerThread.run(WorkerThread.java:561)

09:24:49,820 DEBUG (XNIO-1 task-13) [io.undertow.request.security] <SecurityContextImpl.java:247> Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.CachedAuthenticatedSessionMechanism@18e4682 for /servletContext/2
09:24:49,825 DEBUG (XNIO-1 task-13) [io.undertow.request.security] <SecurityContextImpl.java:110> Authentication result was ATTEMPTED for /servletContext/2


```

File: core/src/main/java/io/undertow/client/http/HttpClientConnection.java [hunk 1]
```diff
@@ -79,11 +79,6 @@
 import java.util.concurrent.CopyOnWriteArrayList;
 
 import static io.undertow.client.UndertowClientMessages.MESSAGES;
-import static io.undertow.util.Headers.CLOSE;
-import static io.undertow.util.Headers.CONNECTION;
-import static io.undertow.util.Headers.CONTENT_LENGTH;
```
Comment (stuartwdouglas) [inline core/src/main/java/io/undertow/client/http/HttpClientConnection.java]: Why did you remove all the static imports?

Comment (frapex) [reply core/src/main/java/io/undertow/client/http/HttpClientConnection.java]: These static imports are no longer required. For example, CLOSE is accessible as Headers.CLOSE.

Comment (stuartwdouglas) [reply core/src/main/java/io/undertow/client/http/HttpClientConnection.java]: Ah, I just got confused because there were not replaced with the equivalent static imports, my bad.
